### PR TITLE
VB-1382   VSIP service to return list of supported prisons

### PR DESCRIPTION
**Pull request by AledWEvans**
## What does this pull request do?

adds to api the ability to get the supported prisons

## What is the intent behind these changes?

to allow the UI to know what can be supported by the VSiP system

### CODE REVIEW

**Comment by tpmcgowan** [comment]
@AledWEvans it looks like this is only returning the prison IDs? We could do with prison ID **and** name - i.e. 'HEI' and 'Hewell (HMP)'

**Comment by rj-adams** [comment]
> 

I think thats fine - if a front end wants to know the description/name for a code - look it up in the source of truth - Prison API for prison names

**Comment by AledWEvans** [comment]
@tpmcgowan you need to state these details on the storey / tickets, This should have happened at refinement and should be discussed then... Also agree with @richpjames the text itself is a UI concern but up for that discussion. The server side dose not have the prison names, the prison id/ names for-fill the storey and task as stated.

**Comment by AledWEvans** [comment]
I'm going to change this after discussion with @dhirajshettymoj and have a separate table!

**Comment by rj-adams** [comment]
It should maybe be in a separate service -it does not really belong to the Visit Scheduler microservice - its more of a service level thing

**Comment by AledWEvans** [comment]
What was decided in planning? there is no task for a new micro service! @rj-adams I will give you a call!

**Comment by tpmcgowan** [comment]
> If a front end wants to know the description/name for a code, it should look it up in the source of truth - Prison API for prison names...

^ fair enough 🙂 - looks like https://github.com/ministryofjustice/prison-register is the place to go for prison names so will look into that...

**Comment by AledWEvans** [comment]
prison names for prison register sounds.. correct!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/session/SessionPrisonWing.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ import javax.persistence.Table
 data class SessionPrisonWing(
 
   @Column(nullable = false)
-  val prisonId: String,
+  val prisonId: Long,
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/session/SessionPrisonWing.kt:16]: will remove this code in separate review and then merge that into this branch for this review!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/SessionTemplateRepository.kt [hunk 2]
```diff
@@ -12,12 +12,12 @@ interface SessionTemplateRepository : JpaRepository<SessionTemplate, Long> {
 
   @Query(
     "select u from SessionTemplate u " +
-      "where u.prisonId = :prisonId " +
+      "where u.prison.code = :prisonCode " +
       "and (u.validToDate is null or u.validToDate >= :firstBookableDay) " +
       "and (u.validFromDate <= :lastBookableDay)"
   )
   fun findValidSessionTemplatesByPrisonId(
```
[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/SessionTemplateRepository.kt:19]: Will do

File: src/main/resources/logback-spring.xml [hunk 3]
```diff
@@ -25,36 +25,10 @@
...
-    </logger>
-
-    <logger name="org.springframework.boot" additivity="false" level="INFO">
-        <appender-ref ref="logAppender"/>
-        <appender-ref ref="consoleAppender"/>
-    </logger>
 
     <root level="INFO">
         <appender-ref ref="logAppender"/>
+        <appender-ref ref="consoleAppender"/>
```
[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/resources/logback-spring.xml:31]: good spot will revert

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/PrisonRepository.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.stereotype.Repository
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.Prison
+
+@Repository
+interface PrisonRepository : JpaRepository<Prison, Long> {
+  fun findByCode(prisonCode: String): Prison
+
+  @Query(
+    "select p.code from Prison p " +
+      "where p.active = true order by p.code"
```
[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/PrisonRepository.kt:14]: Yes only active a prison will only be active when migrating

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt [hunk 5]
```diff
@@ -63,10 +65,13 @@ class VisitService(
 
   fun reserveVisitSlot(bookingReference: String = "", reserveVisitSlotDto: ReserveVisitSlotDto): VisitDto {
 
+    val prison = prisonRepository.findByCode(reserveVisitSlotDto.prisonCode)
```
[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt:68]: same as above this is not optional so will throw error, actually will test this assumption using an integration test... and adjust if need be!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/ConfigService.kt [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.visitscheduler.repository.PrisonRepository
+
+@Service
+@Transactional
+class ConfigService(
+  private val prisonRepository: PrisonRepository,
+) {
+
+  fun getSupportedPrisons(): List<String> {
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/ConfigService.kt:13]: we could add the  @Transactional(readOnly = true) annotation to this method

Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/ConfigService.kt:13]: Good call

**Review by dhirajshettymoj — DISMISSED** [review]
Looks good. But do not merge till we push the current changes into Prod.

**Review by dhirajshettymoj — APPROVED** [review]
LGTM!

**Pull request merged**